src/main/java/com/metamx/metrics/cgroups/CpuAcct.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2017 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.metamx.metrics.cgroups;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.metamx.common.StringUtils;
+import com.metamx.common.logger.Logger;
+import com.metamx.metrics.CgroupUtil;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.LongStream;
+
+public class CpuAcct
+{
+  private static final Logger LOG = new Logger(CpuAcct.class);
+  private static final String CGROUP = "cpuacct";
+  private static final String CGROUP_ACCT_FILE = "cpuacct.usage_all";
+  private static final long[] EMPTY_LONG = new long[0];
+  private static final CpuAcctMetric EMPTY_METRIC = new CpuAcctMetric(EMPTY_LONG, EMPTY_LONG);
+
+  public static CpuAcctMetric parse(final List<String> input)
+  {
+    final int ncpus = input.size() - 1;
+    final long[] usr_time = new long[ncpus];
+    final long[] sys_time = new long[ncpus];
+    for (int i = 1; i < input.size(); ++i) {
+      final String[] splits = input.get(i).split(CgroupUtil.SPACE_MATCH, 3);
+      if (splits.length != 3) {
+        throw new RuntimeException(StringUtils.safeFormat("Error parsing [%s]", input.get(i)));
+      }
+      final int cpu_num = Integer.parseInt(splits[0]);
+      usr_time[cpu_num] = Long.parseLong(splits[1]);
+      sys_time[cpu_num] = Long.parseLong(splits[2]);
+    }
+    return new CpuAcctMetric(usr_time, sys_time);
+  }
+
+  private final CgroupDiscoverer cgroupDiscoverer;
+  private final PidDiscoverer pidDiscoverer;
+
+  public CpuAcct(CgroupDiscoverer cgroupDiscoverer, PidDiscoverer pidDiscoverer)
+  {
+    this.cgroupDiscoverer = cgroupDiscoverer;
+    this.pidDiscoverer = pidDiscoverer;
+  }
+
+  public CpuAcctMetric snapshot()
+  {
+    final File cpuacct = new File(cgroupDiscoverer.discover(CGROUP, pidDiscoverer.getPid()).toFile(), CGROUP_ACCT_FILE);
```

Comment:
We can have another project which might depend on `CpuAcctDeltaMonitor` and has tests which are run locally. Local OS might be not linux.